Sources/Models/Spec.swift

```diff
@@ -39,37 +32,24 @@ struct Spec: Decodable {
     struct RemoteDependency: Decodable {
         let name: String
         let url: String?
-        let version: String?
-        let revision: String?
-        let branch: String?
+        let ref: Ref?
 
         enum CodingKeys: CodingKey {
             case name
             case url
-            case version
-            case revision
-            case branch
         }
 
         init(from decoder: Decoder) throws {
             let container = try decoder.container(keyedBy: CodingKeys.self)
             self.name = try container.decode(String.self, forKey: .name)
             self.url = try container.decodeIfPresent(String.self, forKey: .url)
-            self.version = try container.decodeIfPresent(String.self, forKey: .version)
-            self.revision = try container.decodeIfPresent(String.self, forKey: .revision)
-            self.branch = try container.decodeIfPresent(String.self, forKey: .branch)
+            self.ref = (try? Ref(from: decoder)) ?? nil
```

Comment:
why do we need `?? nil` if `try?` already returns `nil`(nil ?? nil)

Reply:
indeed why 🤣